### PR TITLE
Terrain types

### PR DESCRIPTION
**Pull request by Hop311**
- Moved terrain bmp loading to this repo
- `terrain.txt` loading + parsing
- Terrain type modifier calculation based on number of pixels of each type in the province
- Added simple terrain mapmode

### CODE REVIEW

File: src/openvic-simulation/GameManager.cpp [hunk 1]
```diff
@@ -124,16 +132,23 @@ bool GameManager::expand_building(Province::index_t province_index, const std::s
 	return province->expand_building(building_type_identifier);
 }
 
+static constexpr colour_t LOW_ALPHA_VALUE = float_to_alpha_value(0.4f);
+static constexpr colour_t HIGH_ALPHA_VALUE = float_to_alpha_value(0.7f);
+static constexpr colour_t LAND_COLOUR = 0x0D7017;
+static constexpr colour_t WATER_COLOUR = 0x4287F5;
```
Comment (CptAlanSmith) [inline src/openvic-simulation/GameManager.cpp:138]: Can you comment why these values please? I did google and found nothing obvious.

File: src/openvic-simulation/Modifier.cpp [hunk 2]
```diff
@@ -138,4 +161,4 @@ namespace OpenVic { //so the compiler shuts up
 		}
 		return stream;
 	}
-}
\ No newline at end of file
+}
```
Comment (CptAlanSmith) [inline src/openvic-simulation/Modifier.cpp:164]: Advise minor reduction in nesting if possible.

File: src/openvic-simulation/map/Map.cpp [hunk 3]
```diff
@@ -504,6 +403,134 @@ bool Map::load_region_file(ast::NodeCPtr root) {
 	return ret;
 }
 
+static constexpr colour_t colour_at(uint8_t const* colour_data, int32_t idx) {
+	idx *= 3;
+	return (colour_data[idx + 2] << 16) | (colour_data[idx + 1] << 8) | colour_data[idx];
+}
+
```
Comment (CptAlanSmith) [inline src/openvic-simulation/map/Map.cpp:410]: Please comment these numbers in use. Particularly idx *= 3;
It's not clear of the why, though I can see an 8 and 16 division of data. 24 bit color? I don't know.

**Review by CptAlanSmith — APPROVED** [review]

File: src/openvic-simulation/Modifier.cpp (hunk 2 above)
Comment (Hop311) [reply src/openvic-simulation/Modifier.cpp:164]: This function is going to be rewritten to use a better system than `default_callback`.

**Review by CptAlanSmith — APPROVED** [review]

**Pull request merged**